tensorflow_quantum/python/layers/circuit_executors/input_checks.py

```diff
@@ -0,0 +1,157 @@
+            These are the symbols parameterizing the input circuits.
+        symbol_values: a Python `list`, `tuple`, or `numpy.ndarray` of
+            floating point values, or `tf.Tensor` of dtype `float32`.
+
+    Returns:
+        inputs: `tf.Tensor` of dtype `string` with shape [batch_size]
+            containing the serialized circuits to be executed.
+        symbol_names: `tf.Tensor` of dtype `string` with shape [n_params], which
+            is used to specify the order in which the values in
+            `symbol_values` should be placed inside of the circuits.
+        symbol_values: `tf.Tensor` of dtype `float32` with shape
+            [batch_size, n_params] specifying parameter values to resolve
+            into the circuits following the ordering dictated by `symbol_names`.
+    """
+    # inputs is the circuit(s).
+    symbols_empty = False
+    if symbol_names is None:
+        symbol_names = []
+    if symbol_values is None:
+        symbols_empty = True
+        symbol_values = [[]]
+
+    # Ingest and promote symbol_names.
+    if isinstance(symbol_names, (list, tuple, np.ndarray)):
+        if isinstance(symbol_names, np.ndarray):
+            symbol_names = symbol_names.tolist()
+        if symbol_names and not all(
+            [isinstance(x, (str, sympy.Symbol)) for x in symbol_names]):
+            raise TypeError("Each element in symbol_names"
+                            " must be a string or sympy.Symbol.")
+        symbol_names = [str(s) for s in symbol_names]
+        if not len(symbol_names) == len(list(set(symbol_names))):
+            raise ValueError("All elements of symbol_names must be unique.")
+        symbol_names = tf.convert_to_tensor(symbol_names,
+                                            dtype=tf.dtypes.string)
+    elif tf.is_tensor(symbol_names):
+        if not symbol_names.dtype == tf.dtypes.string:
+            raise TypeError("symbol_names tensor must have dtype string.")
+        if not symbol_names.shape[0] == len(list(set(symbol_names.numpy()))):
+            raise ValueError("All elements of symbol_names must be unique.")
+    else:
+        raise TypeError("symbol_names must be list-like.")
+
+    # Ingest and promote symbol_values.
+    if isinstance(symbol_values, (list, tuple, np.ndarray)):
+        symbol_values = tf.convert_to_tensor(symbol_values,
+                                             dtype=tf.dtypes.float32)
+    elif tf.is_tensor(symbol_values):
+        if not symbol_values.dtype == tf.dtypes.float32:
+            raise TypeError("symbol_values tensor must have dtype float32.")
+    else:
+        raise TypeError("symbol_values must be list-like.")
+
+    # Ingest and promote circuit.
+    symbol_batch_dim = tf.gather(tf.shape(symbol_values), 0)
+    if isinstance(inputs, cirq.Circuit):
+        # process single circuit.
+        inputs = tf.tile(util.convert_to_tensor([inputs]), [symbol_batch_dim])
+    # TODO(zaqqwerty): util.convert_to_tensor does not accept ndarrays
```

Comment:
This is actually intended behavior. ndarrays containg cirq circuits will tend to partially unpack the circuit in weird ways since circuits are iterable.